ts/src/test/Exchange/test.createOrder.ts

```diff
@@ -0,0 +1,228 @@
+        const isClosedFetched = testSharedMethods.confirmOrderState (exchange, buyOrder_nonFillable_fetched, 'closed');
+        assert (!isClosed, logPrefix + ' order should not be filled, but it is. ' + JSON.stringify (buyOrder_nonFillable));
+        assert (!isClosedFetched, logPrefix + ' order should not be filled, but it is. ' + JSON.stringify (buyOrder_nonFillable_fetched));
+        // cancel the order
+        await testCreateOrder_cancelOrder (exchange, symbol, buyOrder_nonFillable['id']);
+        verboseOutput (exchange, symbol, 'SCENARIO 1 PASSED !!!');
+    } catch (e) {
+        throw new Error (logPrefix + ' ' + method + ' failed for Scenario 1: ' + e.toString ());
+    }
+    // *********** [Scenario 1 - END ] *********** //
+
+
+    // ******************************************* //
+    // ************** [Scenario 2] *************** //
+    // ******************************************* //
+    try {
+        // create limit/market order which IS GUARANTEED to have a fill (full or partial), then sell the bought amount
+        const limitBuyPrice_fillable = bestAsk * limitPriceSafetyMultiplierFromMedian;
+        const finalAmountToBuy = getMinimumAmountForLimitPrice (exchange, market, limitBuyPrice_fillable);
+        const buyOrder_fillable = await testCreateOrder_submitSafeOrder (exchange, symbol, 'limit', 'buy', finalAmountToBuy, limitBuyPrice_fillable, {}, skippedProperties);
+        // try to cancel remnant (if any) of order
+        await testCreateOrder_tryCancelOrder (exchange, symbol, buyOrder_fillable, skippedProperties);
+        // now, as order is closed/canceled, we can reliably fetch the order information
+        const buyOrder_filled_fetched = await testSharedMethods.tryFetchOrder (exchange, symbol, buyOrder_fillable['id'], skippedProperties);
+        // we need to find out the amount of base asset that was bought
+        assert (buyOrder_filled_fetched['filled'] !== undefined, logPrefix + ' ' +  exchange.id + ' ' + symbol + ' order should be filled, but it is not. ' + exchange.json (buyOrder_filled_fetched));
+        const amountToSell = buyOrder_filled_fetched['filled'];
+        // if (buyOrder_filled_fetched['filled'] === undefined) {
+        //     // if it was not reported, then the last step is to re-fetch balance and sell all target tokens, subtract the initial balance from the current balance to find out the amount of base asset that was bought
+        //     const balance = await exchange.fetchBalance ();
+        //     amountToSell = balance[market['base']]['free'] - ((initialBaseBalance !== undefined) ? initialBaseBalance : 0);
+        // }
+        //
+        // We should use 'reduceOnly' to ensure we don't open a margin-ed position accidentally (i.e. on FTX you can open a margin position with sell order even if you don't have target base coin to sell)
+        let params = { 'reduceOnly': true };
```

Comment:
you sure this is a bug? or are you trying to use reduceOnly in a spot order?

Reply:
in binance it was a bug by that time, i don't remind if it was fixed later, i have also forgotten to submit that

Reply:
ok, removed and seems that bug was fixed